pyneuroml/swc/ExportNML.py

```diff
@@ -0,0 +1,501 @@
+
+        :param this_point: The current SWCNode.
+        :param cable_id: The ID of the current cable.
+        :param this_type: The type of the current point.
+        :param fract_along_parent_cable: The fractional position along the parent cable.
+        :param version: The NeuroML version being used.
+        """
+        print(
+            f"Handling new cable: Point {this_point.id}, Cable {cable_id}, Type {this_type}"
+        )
+        if version.startswith("1"):
+            fract_info = (
+                f' fract_along_parent="{fract_along_parent_cable}"'
+                if fract_along_parent_cable != 1
+                else ""
+            )
+            self.group_content.append(
+                f'      <cable id="{cable_id}" name="{self.segment_name(this_type)}_{self.seg_per_typ[this_type]}"{fract_info}>'
+            )
+            for group in self.get_groups_for_type(this_point):
+                self.group_content.append(f"        <meta:group>{group}</meta:group>")
+            self.group_content.append("      </cable>")
+        else:
+            for group in self.get_groups_for_type(this_point):
+                if group not in self.segment_groups:
+                    self.segment_groups[group] = []
+                self.segment_groups[group].append(f"{self.cable_prefix_v2}{cable_id}")
+                print(f"Added cable {cable_id} to group {group}")
+
+        self.seg_per_typ[this_type] += 1
+
+    def point_append(self, p: SWCNode, dz: float, loc: str, segment: List[str]):
+        """
+        Append a point to the segment content.
+
+        :param p: The SWCNode to append.
+        :param dz: The z-offset to apply.
+        :param loc: The location identifier ('proximal' or 'distal').
+        :param segment: The list of segment content strings.
+        """
+        print(
+            f"Appending {loc} point: x={p.x:.6f}, y={p.y:.6f}, z={p.z + dz:.6f}, diameter={2 * p.radius:.6f}"
+        )
+        segment.append(
+            f'        <{loc} x="{p.x:.6f}" y="{p.y:.6f}" z="{p.z + dz:.6f}" diameter="{2 * p.radius:.6f}"/>'
+        )
+
+    def get_groups_for_type(self, p: SWCNode) -> List[str]:
+        """
+        Get the groups associated with a given point type.
+
+        :param p: The SWCNode to get groups for.
+        :return: A list of group names.
+        """
+        print(f"Getting groups for point type {p.type}")
+        groups = ["all"]
+        type_groups = {
+            SWCNode.SOMA: ["soma_group", "color_white"],
+            SWCNode.AXON: ["axon_group", "color_grey"],
+            SWCNode.BASAL_DENDRITE: ["basal_dendrite", "dendrite_group", "color_green"],
+            SWCNode.APICAL_DENDRITE: [
+                "apical_dendrite",
+                "dendrite_group",
+                "color_magenta",
+            ],
+        }
+        groups.extend(
+            type_groups.get(p.type, [f"SWC_group_{p.type}", "dendrite_group"])
```

Comment:
what are we doing here? if a group for the type is not found, we're using `SWC_group_{type}` and `dendrite_group`? Not sure if this is correct. If it doesn't belong to the standard types, it shouldn't be valid SWC :thinking: